src/app/pages/SearchBar.tsx

```diff
@@ -0,0 +1,214 @@
+            color: theme.palette.text.hint,
+        },
+        searchResult: {
+            '&:hover': {
+                cursor: 'pointer',
+                color: theme.palette.text.secondary,
+            },
+            '& p, & h6': {
+                marginBottom: theme.spacing(1),
+            },
+        },
+        searchResultPath: {
+            color: theme.palette.text.hint,
+        },
+        searchResultDivider: {
+            margin: `${theme.spacing(3)}px 0`,
+        },
+    })
+);
+
+export const SearchBar: React.FC<SearchbarProps> = (props) => {
+    const classes = useStyles();
+    const onSearch = useSelector((state: AppState) => state.app.onSearch);
+    const dispatch = useDispatch();
+    const [query, setQuery] = useState('');
+    const [searchResults, setSearchResults] = useState([]);
+    const [showSearchResult, setShowSearchResult] = useState(false);
+    const [inputString, setInputString] = useState('');
+    const history = useHistory();
+
+    React.useEffect(() => {
+        if (query) {
+            const loadSearchResults = async (): Promise<void> => {
+                const siteMapDB = await getSitemapDatabase();
+                const indexDB = await getIndexDatabase();
+                if (!siteMapDB || !indexDB) return;
+                // TODO
+                // @ts-ignore
+                setSearchResults(search(query, siteMapDB, indexDB));
+                setShowSearchResult(true);
+            };
+            loadSearchResults();
+        } else {
+            setShowSearchResult(false);
+        }
+    }, [query]);
+
+    // do auto suggestion stuff here
+    const onChangeHandler = (q: string): void => {
+        setInputString(q);
+        if (!q) {
+            setShowSearchResult(false);
+        }
+    };
+
+    const dismissSearchBar = (): void => {
+        setShowSearchResult(false);
+        setQuery('');
+        setInputString('');
+        setSearchResults([]);
+        dispatch({ type: TOGGLE_SEARCH, payload: false });
+    };
+
+    const getSearchResultCountText = (): string => {
+        switch (searchResults.length) {
+            case 0:
+                return `No result found for "${query}".`;
+            case 1:
+                return `${searchResults.length} result found.`;
+            default:
+                return `${searchResults.length} results found.`;
+        }
+    };
+
+    return (
+        <>
+            <Backdrop
+                open={onSearch}
+                transitionDuration={200}
+                className={classes.backdrop}
+                onClick={(): void => {
+                    dispatch({ type: TOGGLE_SEARCH, payload: false });
+                }}
+            />
+
+            {showSearchResult && (
+                <div className={classes.searchResultsOverlay}>
+                    <div className={classes.searchResultsContainer}>
+                        <Typography variant={'body1'} className={classes.searchResultCount}>
```

Comment:
Alignment of the empty counter text seems off:
![image](https://user-images.githubusercontent.com/29152776/83523302-fbdcb900-a4af-11ea-8910-740991c63777.png)
Maybe text align center if we have zero results

Reply:
I don't resonate with center-aligned text in this case